firmware/src/buzzer.c

```diff
@@ -0,0 +1,56 @@
+#include "buzzer.h"
+
+ClockStatus Buzzer_Init(Buzzer *self)
+{
+    self->is_active = false;
+    self->volume = MIN_VOLUME;
+
+    if (self->startPwm == NULL
+        || self->stopPwm == NULL
+        || self->setDutyCycle == NULL) {
+        return CLOCK_FAIL;
+    }
+
+    return CLOCK_OK;
+}
+
+ClockStatus Buzzer_Start(Buzzer *self)
+{
+    if (!self->is_active) {
+        self->startPwm();
+        self->is_active = true;
+    }
+    return CLOCK_OK;
+}
+
+ClockStatus Buzzer_Stop(Buzzer *self)
+{
+    if (self->is_active) {
+        self->stopPwm();
+        self->is_active = false;
+    }
+    return CLOCK_OK;
+}
+
+ClockStatus Buzzer_SetVolume(Buzzer *self, uint8_t vol)
+{
+    if (vol < MIN_VOLUME || vol > MAX_VOLUME) {
+        return CLOCK_FAIL;
+    }
+
+    (self->is_active) ? self->stopPwm() : 0;

```

Comment:
Does this work the same way as an if statement?

Reply:
Yes

Reply:
Is there a performance reason for writing it like this? If you are wanting to keep it to one line I think `if(self->is_active){self->stopPwm();}` is more readable

Reply:
No performance reason, just readability purposes. Cool, will make this change.